### PR TITLE
Add parameter to configure content limit for ingest api

### PR DESCRIPTION
**Pull request by kamalesh0406**
### Description

Describe the proposed changes made in this PR.

This change adds the ability for the user to configure the content limit for the ingest api. It is related to the feature request made in #3797.

### How was this PR tested?

I have tried adding unit tests but they were not working as expected. I wanted the PR out to get the initial comments, I will add the tests in a follow-up commit to this PR.

### CODE REVIEW

File: quickwit/quickwit-config/src/node_config/mod.rs [hunk 1]
```diff
@@ -186,6 +187,7 @@ impl Default for IngestApiConfig {
             max_queue_memory_usage: Byte::from_bytes(2 * 1024 * 1024 * 1024), /* 2 GiB // TODO maybe we want more? */
             max_queue_disk_usage: Byte::from_bytes(4 * 1024 * 1024 * 1024), /* 4 GiB // TODO maybe we want more? */
             replication_factor: 1,
+            content_length_limit: 10 * 1024 * 1024 /* 10 MB */
```
Comment (ChillFish8) [inline quickwit/quickwit-config/src/node_config/mod.rs:190]: `10 << 20` would probably be better, but I think 10MB is too low that would be about 250ms worth of data 1/4 of the data a single pipeline could ingest. 100MB might be more suitable.

**Review by fulmicoton — CHANGES_REQUESTED** [review]
Can we add a unit test checking we reject payload above the limit configured?
A test on the filter level is ok.

File: quickwit/quickwit-config/src/node_config/mod.rs (hunk 1 above)
Comment (kamalesh0406) [reply quickwit/quickwit-config/src/node_config/mod.rs:190]: Okay I will modify the default value to 100MB.

**Comment by kamalesh0406** [comment]
> Can we add a unit test checking we reject payload above the limit configured? A test on the filter level is ok.

Sure I was working on the unit test I will push it when its done.

File: quickwit/quickwit-config/src/node_config/mod.rs (hunk 1 above)
Comment (guilload) [reply quickwit/quickwit-config/src/node_config/mod.rs]: 10 was perfectly fine. Our gRPC stack does not allow messages bigger than 4 MB at the moment. Not to mention rate limiting only applies after ingest payloads are already in memory.

Comment (kamalesh0406) [reply quickwit/quickwit-config/src/node_config/mod.rs]: I agree with @guilload as well, I have reverted it back to 10MB.

**Review by fulmicoton — APPROVED** [review]

**Comment by fulmicoton** [comment]
Thank you @kamalesh0406 !

**Pull request merged**